packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -1338,9 +1344,21 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
         );
     }
 
-    final BoxDecoration? effectiveDecoration = widget.decoration?.copyWith(
-      border: resolvedBorder,
-      color: enabled ? decorationColor : disabledColor,
+    // If the text field should have a border, set the background color to the
+    // disabled color only if a color for the box decoration was not set.
+    final BoxDecoration? effectiveDecoration = (widget._textFieldType == _TextFieldType.bordered
+      ? widget.decoration?.copyWith(
+          border: resolvedBorder,
+          color: enabled ? decorationColor
```

Comment:
Maybe we should check if this color is a `WidgetStateColor` similar to how we've done elsewhere for `MaterialStateColor` in the past: https://api.flutter.dev/flutter/material/TimePickerThemeData/dayPeriodColor.html.

Then they could customize it for all states easier. It'd need to be called out in the documentation.

Reply:
From what I can find, `CupertinoTextField` only has a state for when it is enabled or disabled. I can't find information on any other states (hovering, pressed, focused, etc). So would this be relevant in this case?

It might be worthwhile to create another PR to introduce widget states to `CupertinoTextField`, at which point this might be more relevant.

Reply:
The cupertino library uses `CupertinoDynamicColor` which is a similar system but with potentially different input (resolveWith parameters).